src/test/java/schools/SchoolTest.java

```diff
@@ -27,76 +25,110 @@ public void clearDataAfterTest(){
+     *
+     * Jakiekolwiek sa odpowiedzi na ww pytania powinny byc dla nich testy.
+     */
     @Test
-    public void shouldReturnFalseIfClassExist(){
+    // dobrze by nazwy reprezentowaly efekt biznesowy a nie techniczny. Oczywiscie czasem sie nie da ale akurat tutaj
+    // jest mozliwosc opisania efektu biznesowego
+    public void shouldNotLetAddTheSameClassMoreThanOne(){
         // given
         school.addClass(clazzA);
+        // jak bedzie taka operacja robiona na klasach to musimy je zwrocic
+        // a z opisu w klasie School nie wynika, ze to jest oczekiwane
+
+        // assertThat(school.howManyClasses()).isEqual(1);
 
         // when
         boolean result = school.addClass(clazzA);
 
         // then
         assertFalse(result);
+        // assertThat(school.howManyClasses()).isEqual(1);
     }
     @Test
-    public void shouldReturnTrueIfClassCanBeAdded(){
+    public void shouldAddNewClassToSchool(){
         // given
         school.addClass(clazzA);
+        // assertThat(school.howManyClasses()).isEqual(0);
 
         // when
+        // to nie jest najszczesliwsza logika
+        // lepiej by bylo dodac
+        // school.hasClass(clazzA);
         boolean result = school.getSchoolClasses().containsValue(clazzA);
 
         // then
+
+        // to sprawdzenie jest ok
         assertTrue(result);
         assertThat(school.getSchoolClasses()).hasSize(1);
     }
     @Test
-    public void shouldReturnTrueIfRemovingSchoolClassSuccessful(){
+    public void shouldRemoveExistingClass(){
         // given
         school.addClass(clazzA);
 
         // when
+        // o nie jest test tego co bylo opisane wyzej - to jest nadal element given i lepiej by bylo zrobic to przez
+        // hasClass
         assertThat(school.getSchoolClasses()).containsValue(clazzA);
 
         // then
-        assertTrue(school.removeClass(clazzA));
+        // to jest when i lepiej by wygladalo
+        var result = school.removeClass(clazzA);
+
+        // then wlasciwe - do poprawy asercja na assertThat()
+        assertTrue(result);
         assertThat(school.getSchoolClasses()).hasSize(0);
     }
     @Test
-    public void shouldReturnFalseIfRemovingSchoolClassDoNotExist(){
+    public void shouldNotRemoveClassIfNotInSchoolAndLetContinue(){
         // given
-        school.addClass(clazzA);
-        assertThat(school.getSchoolClasses()).containsValue(clazzA);
-        school.removeClass(clazzA);
+        // tu bym nic nie robila, bo wtedy na pewno szkola nie ma klasy
 
         // when
-        assertThat(school.getSchoolClasses().size()).isEqualTo(0);
+        var result = school.removeClass(clazzA);
 
         // then
-        assertFalse(school.removeClass(clazzA));
+        assertFalse(result);
     }
     @Test
     public void shouldReturnTotalSchoolHeadcount(){
         // given
         school.addClass(clazzA);
 
         // when
+        // cos tutaj jest nie w porzadku - getHeadcount nie powinna miec argumentu
```

Comment:
metoda w klasie School posiada argument, dlatego też jest i tu; sygnatura metody: `
public int getHeadcount(Map<String, SchoolClass> schoolClasses)  `{}

Reply:
tak zauwazylam, ze ma. I to mnie zastanawia. Jakby ta metoda byla w innej klasie, takiej jaka ma dodatkowe operacje na szkole, to jej argument bylby zrozumialy. Ale tutaj klasa School wie jakie sa w niej klasy wiec nie jest to potrzebne. CO wiecej zobacz co masz school.getHeadcount - czyli operujesz na school a potem w parametrze masz to samo school. getSchoolClasses. Czy nadal operujesz na dokladnie tej samej school (w obu jest school.) To jest bardzo mocny sygnal, ze cos jest nie tak z ta metoda. Jezeli zostaje w klasie school - nie powinna miec parametry. Jezeli przenosisz do klasy narzedziowej moze miec ale wtedy nie bedzie testu dla niej tutaj a w tescie tej nowej klasy.
To jest bardzo dobry przyklad tego co sygnalizowalam na spotkaniu - testy nie tylko pozwalaja sprawdzic czy kod dziala ale rowniez "pokazuje", ze uklad klas(y) niekoniecznie jest najlepszy